src/lib.rs

```diff
@@ -471,25 +471,25 @@ impl<Types: IpfsTypes> Ipfs<Types> {
     /// Subscribes to a given topic. Can be done at most once without unsubscribing in the between.
     /// The subscription can be unsubscribed by dropping the stream or calling
     /// [`pubsub_unsubscribe`].
-    pub async fn pubsub_subscribe(&self, topic: &str) -> Result<SubscriptionStream, Error> {
+    pub async fn pubsub_subscribe(&self, topic: String) -> Result<SubscriptionStream, Error> {
         let (tx, rx) = oneshot_channel();
 
         self.to_task
             .clone()
-            .send(IpfsEvent::PubsubSubscribe(topic.into(), tx))
+            .send(IpfsEvent::PubsubSubscribe(topic, tx))
             .await?;
 
         rx.await?
-            .ok_or_else(|| format_err!("already subscribed to {:?}", topic))
+            .ok_or_else(|| format_err!("already subscribed to the given topic"))
```

Comment:
I usually do error messages in `constant_part: variable_part` or "like [perror(3)](https://linux.die.net/man/3/perror)" but here I failed with it apparently.

I am still very much in favor of including more the topic in this error for example. Errors like this are quite dreadful to have happen in production, and then even the stacktrace doesn't help at all, since the error comes from the internal state which the user may have been modifying concurrently by dropping the previous receiver, but it all happens in quite unexplainable order because of the many queues in play.

Reply:
More examples for "Context for errors" can be found related to this: https://github.com/dvc94ch/rust-ipld/pull/1 which was adding context to quite strange error there was for a incorrectly formatted Cid (`b"0".concat(cid.to_bytes())`), it was a bit tricky to track down.